### DISCUSSION TITLE
May I ask how to obtain the output of each layer and the activation values when i use JAX?

### DISCUSSION THREAD

**Discussion by sglucas** [google/flax#3199] [Q&A]
May I ask how to obtain the output of each layer and the activation values when i use JAX with flax? I hope you could provide a simple example.

Best,
Lucas

**Comment by cgarciae** [comment]
Hey @sglucas, take a look at the [Extracting intermediate values](https://flax.readthedocs.io/en/latest/guides/extracting_intermediates.html) guide.
